client/src/pages/dashboard/views/Outbox.svelte

```diff
@@ -13,83 +13,89 @@
     import SendRow from '../../../components/ui/itemrow/SendRow.svelte';
     import { loadAll } from '@square/svelte-store';
     import { deferredSnaps } from '../stores/DeferredStore.ts';
+    import { Document } from '../../../../../model/src/document.ts';
+
+    interface Context {
+        docId: Document['id'] | null,
+        mode: Status | null,
+        context: boolean,
+    }
+
+    let ctx = null as Context | null;
 
     $: ({ currentOffice } = $dashboardState);
 
-    let showContextMenu = false;
-    let showInsertSnapshot = false;
-    let showCreateDocument = false;
+    function openContext(doc: Document['id']) {
+        ctx = { docId: doc, mode: null, context: true };
+    }
 
-    let insertSnapshotAction = null as Status | null;
-    let currentContext = null as ContextPayload | null;
+    function openInsertSnapshot(doc: Document['id'], mode: Status) {
+        ctx = { docId: doc, mode: mode, context: false };
+    }
 
-    function overflowClickHandler(e: CustomEvent<ContextPayload>) {
-        currentContext = e.detail;
-        showContextMenu = true;
+    function openCreateDocument() {
+        ctx = { docId: null, mode: Status.Register, context: false };
     }
 
-    function contextMenuHandler(e: CustomEvent<ContextPayload>) {
-        switch (e.type) {
-            case Events.SendDocument:
-                insertSnapshotAction = Status.Send;
-                break;
-            case Events.TerminateDocument:
-                insertSnapshotAction = Status.Terminate;
-                break;
-            default: break;
-        }
-        if (currentOffice) showInsertSnapshot = true;
+
+    function resetContext() {
+        ctx = null;
     }
 </script>
-
 {#if currentOffice === null}
     You must select an office before accessing the Outbox page.
 {:else}
     <h1>Outbox</h1>
     
-    <Button on:click={() => (showCreateDocument = true)}>
+    <Button on:click={openCreateDocument.bind(null)}>
         Register and Stage a New Document
     </Button>
 
     {#await loadAll([documentOutbox, deferredSnaps])}
         <p>Loading outbox...</p>
     {:then}
         <h2>Staged Registered Documents</h2>
-        {#each $documentOutbox.ready as entry (entry.doc)}
+        {#each $documentOutbox.ready as {doc, category, creation, title} (doc)}
             <RegisterRow 
-                {...entry}
+                {doc}
+                {category}
+                {title}
+                {creation}
                 iconSize={IconSize.Large} 
-                on:overflowClick = {overflowClickHandler}
+                on:overflowClick = {openContext.bind(null, doc)}
             />
         {/each}
 
         <h2>Sent Documents</h2>
         {#each $documentOutbox.pending as entry (entry.doc)}
-            <SendRow iconSize={IconSize.Large} {...entry} />
+            <SendRow 
+                {...entry}
+                iconSize={IconSize.Large}
+            />
```

Comment:
Surely we have enough screen real estate for this one-liner.
```suggestion
            <SendRow {...entry} iconSize={IconSize.Large} />
```